src/gameComponents/HullZone.java

```diff
@@ -0,0 +1,148 @@
+package gameComponents;
+
+import org.joml.Vector2f;
+
+public class HullZone {
+
+    /*
+    A hull zone is a collection of space on a ship that stores shields attack values and some location data.
+    A hull zone can take shield damage without outside assistance
+     */
+
+    //Available enums for a ship facing. AUX arcs exist for huge ships, otherwise use the normal ones.
+    public enum Facing {FRONT, LEFT, RIGHT, BACK, AUX_LEFT, AUX_RIGHT}
+
+    //The named facing of the ship
+    private Facing facing;
+
+    //The anti-ship battery values
+    private int redBattery;
+    private int blueBattery;
+    private int blackBattery;
+
+    //The current and maximum shield value of the hull zone. In most cases these start identical
+    private int shields;
+    private int shieldsMaximum;
+
+    //The Line of sight offset relative to ship center for a HullZone
+    private Vector2f LoSOffset;
+
+    //The left/right origin offsets relative to ship center for a HullZone
+    private Vector2f leftArcOffset;
+    private Vector2f rightArcOffset;
+
+    //The left right arc angles for a ship.
+    private float leftArcAngle;
+    private float rightArcAngle;
```

Comment:
Not wild about all this geometry related stuff that isn't capitalizing on the BBDGeometry package.

Next step is to probably attach a BBDPolygon to things instead of maintaining lots of values since the BBDPolygon can do things like overlaps, intersections, range measurements, etc.  Those are built in and then remove a lot of geometry related fluff from this class.

Reply:
I anticipate making the Polygon as part of the ship constructor and feeding that along with some parsed values to the hull zone constructor.  That's because we need the ship base to exist so we can use its dimensions to construct the geometry.

Reply:
This will be a later refactor

Reply:
Part of the issue I'm directly facing is that a hull zone has a variety of numbers for how many sides a particular hull zone has, anywhere from 3-6, so there isn't a boring simple thing like "The first and last sides of this polygon are the firing arc lines." At some level we still will end up needing to define which of the lines are arc lines and where they start from to reduce the possibility we measure in the wrong direction.

In the current form, my mindset was largely circling around the idea of minimizing float drift, wherein recalculating our positions <many> times will have adverse effects over the long run. Granted I'm looking at a scenario where a hull zone gets recalcualted up to 25 times. (6 turns of speed 4 bump to 0, and the initial placement) so maybe 25 drifts is not a big deal.

Instead of having a constantly recalculating hull zone, we could instead have a set of known starting points that we could plastered onto a ship later to create "adjusted hull zones" which would be the correctly rotated translated etcetera'd versions of themselves

Reply:
This has about as much drift as a BBDPolygon just because floats.  An "adjusted hull zone" would likely suffer from drift because your base location will drift as it moves.

Also worth pointing out that drifts are on the order of 10^-7ish.  Where 1 is a millimeter.  So drift per operation is on the order of fractions of a nanometer.  We'd have to do a LOT of operations before that is relevant.

Reply:
I think last time I made a run at this project I bit the bullet and declared that all hullzones were 6 points and moved on.  Several of those points were identical, but it made the coding sane.